### PR TITLE
AK: Refactor `JSONParser::consume_and_unescape_string` a bit

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
This code was quite unwieldy and repetitive.
The long `if(next_is('X'))` chain is now a small `switch`. I also reinstated the fast path for long sequences of literal characters, which was partially broken with the introduction of `GenericLexer`.
I also added some spec comments, and implementation notices

---

I know bad commit title, but could not think of something better

### CODE REVIEW

File: AK/JsonParser.cpp [hunk 1]
```diff
@@ -18,105 +18,105 @@ constexpr bool is_space(int ch)
     return ch == '\t' || ch == '\n' || ch == '\r' || ch == ' ';
 }
 
+/* ECMA-404 9 String
+ * Boils down to
+ * STRING = "\"" *("[^\"\\]" | "\\" ("[\"\\bfnrt]" | "u[0-9A-Za-z]{4}")) "\""
+ *     │├── " ──╮───────────────────────────────────────────────╭── " ──┤│
+ *              │                                               │
```
Comment (ADKaster) [inline AK/JsonParser.cpp:25]: wrong comment style

**Review by ADKaster — COMMENTED** [review]
Commit suggestion: AK: Bring JsonParser's string consumption closer to the ECMA 404 spec

**Comment by Hendiadyoin1** [comment]
Changes:
More spec  comments and implementation notices
Actually reference the correct commit which "broke" the reinstated optimization

**Review by ADKaster — APPROVED** [review]
I'll merge this b/c we have existing tests and the refactor looks pretty decent. But I wonder, does this change make it so that the currently-commented-out (for two years already! :catdogalert:) `TEST_CASE(json_utf8_multibyte)` in TestJSON passes?

**Pull request merged**